src/main/java/imsam/mgen/MultiTargetModelGenerator.java

```diff
@@ -0,0 +1,347 @@
+                        rate);
+                    if((!state.transitionOutExists(transition)) && (successor != state.stateId)){
+                        logger.trace("Connecting state " +state.stateId+ " to state "+ successor);
+                        state.transitionsOut.add(transition);
+                        stateSpace[successor].transitionsIn.add(transition);
+                        //add to adjacency matrix
+                        adjacencyMatrix[state.stateId][successor] = rate;
+                    }else{
+                        logger.trace(state.stateId+" already connects to "+successor);;
+                    }
+                }
+            }else{
+                logger.trace("Skipping target state "+ state.stateId);
+            }
+        }
+        /*
+         * Finally to help visualize the generated model, a python script is generated that will draw the graph.
+         * It utilizes the networkx, matplotlib, and graphviz libraries. Total runtime is also calculated.
+         */
+        logger.debug("Generating drawGraph.py");
+        try{
+            processMatrix();
+        }catch(IOException e){
+            e.printStackTrace();
+        }
+        long end = System.nanoTime();
+        logger.info(numberOfStates+"-state model generated in "+(end-start)/1000000+ "ms");
+    }
+
+    /**
+     * Randomly choose target states from state space, arbitrarily bounded at 1/3 of state space
+     * @return List of target states
+     */
+    private List<Integer> generateTargetStates(){
+        StringBuilder strBld = new StringBuilder();
+        List<Integer> targets = new ArrayList<>();
+        double third = 0.3333333333333333f;
+        int targetBound = (numberOfStates==2)?1:(int)(third*numberOfStates); //get bound targets to 1/3 of state space
+        logger.info("Target Bound: "+targetBound);
+        int numberOfTargets = (int)((Math.random()*(targetBound-1))+1); //Randomly choose number of targets, always ensuring at least 1
+        logger.info("Target Size: "+numberOfTargets);
+        logger.debug(numberOfTargets+(numberOfTargets==1?" state has ":" states have ")+"been selected as target(s)");
+        for(int targetIndex = 0; targetIndex < numberOfTargets; targetIndex++){
+            //draw targets
+            int pickTarget = (int)((Math.random()*(numberOfStates-1))+1); 
+            //redraw for unique targets
+            while(targets.contains(pickTarget)){  
+                pickTarget = (int)((Math.random()*(numberOfStates-1))+1);
+            }
+            targets.add(pickTarget);
+            strBld.append(pickTarget+" ");
+        }
+        logger.info("Generated Target States: "+ strBld.toString());
+        return targets;
+    }
+
+    /**
+     * This method generates a python script to draw the generated model.
+     * It utilizes the matplotlib, networkx, and graphviz libraries.
+     */
+    private void processMatrix() throws IOException{
+        String fileName = "drawGraph.py";
```

Comment:
Rather than creating a python file to draw a graph, may as well use a native Java library to do so: https://stackoverflow.com/questions/2981825/drawing-graphs-on-java#2981837

Reply:
Thanks, I will look into this.